facilities/models.py

```diff
@@ -0,0 +1,83 @@
+import random
+from django.db import models
+from common.models import AbstractBase, SubCounty
+
+
+class Owner(AbstractBase):
+    name = models.CharField(max_length=100, unique=True)
+    description = models.TextField(null=True, blank=True)
+    code = models.CharField(max_length=100, unique=True)
+
+    def generate_code(self):
+        return "{}{}".format("OWNER", str(self.id))
+
+    def save(self, *args, **kwargs):
+        if not self.code:
+            self.code = self.generate_code()
+        super(Owner, self).save(*args, **kwargs)
+
+
+class Service(AbstractBase):
+    name = models.CharField(max_length=255)
+    description = models.TextField(null=True, blank=True)
+    code = models.CharField(max_length=100, unique=True)
+
+    def __unicode__(self):
+        return self.name
+
+    def generate_code(self):
+        return "{}{}".format(self.name, str(self.id))
+
+    def save(self, *args, **kwargs):
+        if not self.code:
+            self.code = self.generate_code()
+        super(Service, self).save(*args, **kwargs)
+
+
+class GIS(AbstractBase):
+    latitude = models.CharField(max_length=255)
+    longitude = models.CharField(max_length=255)
+    is_classified = models.BooleanField(default=False)
+
+
+class FacilityStatus(AbstractBase):
+    name = models.CharField(max_length=100)
+
+
+class FacilityType(AbstractBase):
+    name = models.CharField(max_length=100)
+
+
+class Facility(AbstractBase):
+    name = models.CharField(max_length=100, unique=True)
+    code = models.CharField(max_length=100, unique=True)
+    gis = models.OneToOneField(GIS, null=True, blank=True)
```

Comment:
I'd rather have the `GIS` mode have the `OneToOneField` relationship with the `Facility` ( inverse of what you have ).

Then - when you move this field, make it non-nullable.